src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutPresenter.cs

```diff
@@ -1,13 +1,1008 @@
+				for (var i = 0; i < nCount; ++i)
+				{
+					MenuFlyoutItemBase item;
+					MenuFlyoutItem itemAsMenuItem;
+					MenuFlyoutSubItem itemAsMenuSubItem;
+					IconElement iconElement;
+					string keyboardAcceleratorText;
+
+					item = spItems[i] as MenuFlyoutItemBase;
+
+					// To prevent casting the same item more than we need to, each cast is conditional
+					// on the previous one failing. This way we only cast each item as many times as we
+					// need to.
+					itemAsMenuItem = item as MenuFlyoutItem;
+					if (itemAsMenuItem != null)
+					{
+						m_containsToggleItems = m_containsToggleItems || (itemAsMenuItem as MenuFlyoutItem).HasToggle();
+
+						iconElement = (itemAsMenuItem as MenuFlyoutItem).Icon;
+						m_containsIconItems = m_containsIconItems || iconElement != null;
+
+						keyboardAcceleratorText = (itemAsMenuItem as MenuFlyoutItem).KeyboardAcceleratorTextOverride;
+						m_containsItemsWithKeyboardAcceleratorText = m_containsItemsWithKeyboardAcceleratorText || !string.IsNullOrEmpty(keyboardAcceleratorText);
+					}
+					else
+					{
+						itemAsMenuSubItem = item as MenuFlyoutSubItem;
+						if (itemAsMenuSubItem != null)
+						{
+							iconElement = (itemAsMenuSubItem as MenuFlyoutSubItem).Icon;
+							m_containsIconItems = m_containsIconItems || iconElement != null;
+						}
+					}
+
+					if (m_containsIconItems && m_containsToggleItems && m_containsItemsWithKeyboardAcceleratorText)
+					{
+						break;
+					}
+				}
+
+				UpdateTemplateSettings();
+			}
+		}
+
+		// Create MenuFlyoutPresenterAutomationPeer to represent the 
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			return new MenuFlyoutPresenterAutomationPeer(this);
+		}
+
+		void UpdateVisualStateForPlacement(FlyoutBase.MajorPlacementMode placement)
+		{
+			m_mostRecentPlacement = placement;
+
+			UpdateVisualState(false);
+		}
+
+		void ResetVisualState()
+		{
+			VisualStateManager.GoToState(this, "None", false);
+		}
+
+		private protected override void ChangeVisualState(
+		   // true to use transitions when updating the visual state, false
+		   // to snap directly to the new visual state.
+		   bool bUseTransitions)
+		{
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			base.OnApplyTemplate();
+
+			ScrollViewer spScrollViewer;
+
+			// Get the ScrollViewer template part
+			spScrollViewer = this.GetTemplateChild("MenuFlyoutPresenterScrollViewer") as ScrollViewer;
+			m_tpScrollViewer = spScrollViewer;
+
+			// Apply a shadow
+
+			// UNO TODO
+			//bool isDefaultShadowEnabled = IsDefaultShadowEnabled;
+			//if (isDefaultShadowEnabled)
+			//{
+			//	DependencyObject spChild;
+			//	VisualTreeHelper.GetChildStatic(this, 0, &spChild);
+			//	var spChildAsUIE = spChild as UIElement;
+
+			//	if (spChildAsUIE != null)
+			//	{
+			//		ApplyElevationEffect(spChildAsUIE, GetDepth());
+			//	}
+			//}
+		}
+
+		string GetOwnerName()
+		{
+			var parentMenuFlyout = GetParentMenuFlyout();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'parentMenuFlyout'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)